enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,401 @@
+have increased CNI failure rate.
+
+#### Support Procedures
+
+If Multus cannot configure Pod's networking according to the annotations (any of the CNIs fail),
+Pod will not start and its events should contain error from the Multus. For example:
+```
+Warning  NoNetworkFound          0s                 multus             cannot find a network-attachment-definition (asdasd) in namespace (default): network-attachment-definitions.k8s.cni.cncf.io "asdasd" not found
+```
+
+To address such issues, user can:
+- verify values in both NetworkAttachmentDefinitions and Annotations,
+- remove Annotation to verify if the Pod is created successfully with just the default network.
+
+## Implementation History
+
+<!-- Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`. -->
+
+## Alternatives
+
+### Thick plugin architecture
+
+In 2022 Multus was rewritten in manner called "thick plugin" and changes mode of operation significantly.
+Major difference is that DaemonSet is no longer dummy (i.e. Thin: creates kubeconfig, config, copies Multus
+CNI binary to the host, and finally sleeps or watches files to update the kubeconfig and/or config).
+Instead it is the brain of the operation: CNI binary on the host is only a shim that forwards the request
+to the DaemonSet which executes all of the delegates. It also exports a metric, but it was not
+deemed useful.
+
+Decision to use thin instead of thick architecture is mostly driven by resource consumption:
+Multus CNI binary in thin mode only uses resources when it runs and DaemonSet idles (or is close to it),
+whereas thick Multus' DaemonSet is an application that uses resources even if there are no new Pods.
+
+If we would decide to use thick plugin, we still would need to create new image as the one used in OCP
+is not suitable for edge deployments (1.2GB). Another pro for thick plugin is that is has better
+test coverage as it is used in OpenShift. Also, if there will be a new CNI spec that includes UPDATE
+command it thick has better chance of supporting that.
+
+Even though thin plugin suits MicroShift needs better, we should strive toward making as little
+breaking changes as possible compared to OpenShift's thick multus when preparing manifests for
+MicroShift's thin Multus.
+
+See [Multus Thick Plugin](https://github.com/openshift/multus-cni/blob/master/docs/thick-plugin.md).
+
+### Using network plugins from RHEL repositories
+
+RHEL ships network plugins RPM that includes delegate CNIs we aim to support like `bridge`, `macvlan`, etc.
+Originally that RPM was meant for Podman networking, but Podman shipped with RHEL9 does not use them anymore.
+This means they can exist only for compatibility and are not actively maintained.
+On the other hand, OpenShift networking team (with whom we have ongoing cooperation) are actively
+maintaining these binaries and quickly addressing any CVEs or bugs.
```

Comment:
It's important to mention here that the binaries delivered by the networking team come in a container image as part of the OCP payload.